go/signedexchange/signedexchange.go

```diff
@@ -504,3 +507,40 @@ func (e *Exchange) PrettyPrintPayload(w io.Writer) {
 	fmt.Fprintf(w, "payload [%d bytes]:\n", len(e.Payload))
 	w.Write(e.Payload)
 }
+
+func (e *Exchange) JSONPrint(w io.Writer) {
```

Comment:
I feel like this should live in dump-signedexchange, since users who wish to use this Go code as a library is unlikely to need this.

Reply:
Done. (FWIW, I originally put it here because the signature matched that of `PrettyPrintHeaders()`.)